File: basics/variables.cpp
#include <cstddef> //size_t
#include <cstdio>
#include <clocale> //setlocale()

int main(){
    int a = 1;
    const unsigned int b = 2;
    auto c = 0xffff;// -std=c++11
    size_t d = 4;// x bits on x-bit machine
    float x = 3.0; double y = 1e-10;
    wchar_t ch = L'烟'; setlocale(LC_ALL, "");
    printf("%lc %d\n",ch,sizeof(wchar_t));
}

File: basics/variables.rs
use std::mem::size_of; //similar to python import
fn main(){
    let mut a:i32 = 1;
    let b: u32 = 2;
    let c = 0xffff;
    let d: usize = 4;// x bits on x-bit machine
    let x: f32 = 3.0;let y: f64 = 1e-10;
    let ch:char = '烟';//4 byte unicode character
    println!("{} {}",ch,size_of::<char>());
}